In [1]:
import pandas as pd 
import sqlite3 

In [2]:
df = pd.read_excel("Dataset for Data Analytics.xlsx")
df.head()

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [3]:
conn = sqlite3.connect('project3.db')

In [4]:
df.to_sql('Orders', conn, if_exists='replace', index=False)

1200

In [5]:
pd.read_sql(""" SELECT* FROM Orders LIMIT 5 """, conn)

,OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
0,ORD200000,2023-01-04 00:00:00,C72649,Monitor,5,570.62,928 Main St,Debit Card,Shipped,TRK37947903,7,SAVE10,Instagram,2853.10
1,ORD200001,2024-08-23 00:00:00,C75739,Phone,2,151.35,823 Main St,Online,Shipped,TRK91186779,3,SAVE10,Referral,302.70
2,ORD200002,2024-02-27 00:00:00,C81728,Tablet,5,550.68,512 Main St,Credit Card,Cancelled,TRK42903982,8,FREESHIP,Email,2753.40
3,ORD200003,2023-10-15 00:00:00,C33540,Chair,1,273.19,275 Main St,Debit Card,Returned,TRK62788070,5,SAVE10,Facebook,273.19
4,ORD200004,2025-05-08 00:00:00,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04


In [9]:
pd.read_sql("""
SELECT COUNT(*) AS TotalOrders
FROM Orders
""", conn)

,TotalOrders
0,1200


In [12]:
!pip install ipython-sql

In [16]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [18]:
%sql sqlite:///project3.db

In [23]:
%reload_ext sql

In [26]:
%config SqlMagic.style = '_DEPRECATED_DEFAULT'

In [27]:
%%sql
SELECT COUNT(*) AS TotalOrders
FROM Orders;

 * sqlite:///project3.db
Done.


TotalOrders
1200


Observation:
The Orders table contains 1200 records.

Insight:
The dataset consists of 1200 customer orders available for analysis.

In [28]:
%%sql
SELECT AVG(TotalPrice) AS Averageordervalue FROM Orders;

 * sqlite:///project3.db
Done.


Averageordervalue
1053.9683


Insight:
Customers spend approximately ₹1054 per order on average.

In [29]:
%%sql 
SELECT SUM(TotalPrice) AS TotalRevenue FROM Orders;

 * sqlite:///project3.db
Done.


TotalRevenue
1264761.96


Observation:
The total revenue generated from all orders is ₹12,64,761.96.
    
Insight:
The business generated approximately ₹12.65 lakh in revenue from 1200 customer orders.

In [31]:
%%sql 
SELECT Product, COUNT(*) AS TotalOrders FROM Orders
GROUP BY Product;

 * sqlite:///project3.db
Done.


Product,TotalOrders
Chair,178
Desk,170
Laptop,173
Monitor,163
Phone,156
Printer,181
Tablet,179


In [32]:
%%sql
SELECT Product,
COUNT(*) AS TotalOrders
FROM Orders
GROUP BY Product
ORDER BY TotalOrders DESC;

 * sqlite:///project3.db
Done.


Product,TotalOrders
Printer,181
Tablet,179
Chair,178
Laptop,173
Desk,170
Monitor,163
Phone,156


Observation:
Printer has the highest number of orders (181), while Phone has the lowest number of orders (156).

Insight:
Printer is the most popular product among customers, while Phone has the lowest demand in the dataset.

In [33]:
%%sql 
SELECT*FROM Orders 
Where OrderStatus='Delivered';

 * sqlite:///project3.db
Done.


OrderID,Date,CustomerID,Product,Quantity,UnitPrice,ShippingAddress,PaymentMethod,OrderStatus,TrackingNumber,ItemsInCart,CouponCode,ReferralSource,TotalPrice
ORD200004,2025-05-08 00:00:00,C81840,Printer,4,626.01,668 Main St,Online,Delivered,TRK29241424,8,SAVE10,Email,2504.04
ORD200015,2023-07-17 00:00:00,C39416,Printer,1,473.96,942 Main St,Cash,Delivered,TRK54930938,3,None,Google,473.96
ORD200021,2024-11-17 00:00:00,C99023,Monitor,4,342.95,569 Main St,Cash,Delivered,TRK52646983,6,None,Google,1371.8
ORD200024,2024-10-30 00:00:00,C17470,Laptop,2,230.95,956 Main St,Online,Delivered,TRK28775057,3,WINTER15,Referral,461.9
ORD200025,2024-07-24 00:00:00,C42577,Monitor,1,399.98,709 Main St,Debit Card,Delivered,TRK69423327,3,WINTER15,Email,399.98
ORD200026,2024-03-06 00:00:00,C21944,Laptop,5,59.55,296 Main St,Online,Delivered,TRK80276812,6,None,Referral,297.75
ORD200029,2024-09-20 00:00:00,C33260,Desk,3,423.53,930 Main St,Credit Card,Delivered,TRK59744090,6,SAVE10,Referral,1270.59
ORD200032,2023-05-01 00:00:00,C12388,Tablet,5,536.72,830 Main St,Cash,Delivered,TRK99261395,10,None,Email,2683.6
ORD200038,2024-12-19 00:00:00,C15924,Laptop,1,245.89,549 Main St,Debit Card,Delivered,TRK14835975,3,SAVE10,Google,245.89
ORD200039,2025-02-27 00:00:00,C96664,Printer,3,256.46,666 Main St,Credit Card,Delivered,TRK85508960,4,SAVE10,Referral,769.3799999999999


Observation:
The query displays only orders with a Delivered status.
    
Insight:
Delivered orders represent successfully completed customer purchases and can be used to analyze fulfilled sales.

In [34]:
%%sql
SELECT PaymentMethod, COUNT(*) AS TotalOrders FROM Orders
GROUP BY PaymentMethod
ORDER BY TotalOrders DESC;

 * sqlite:///project3.db
Done.


PaymentMethod,TotalOrders
Online,258
Cash,246
Credit Card,234
Debit Card,232
Gift Card,230


Observation:
Online payment has the highest number of orders.
    
Insight:
Customers prefer online payment methods over other payment options.

Final Insights

1. The dataset contains 1200 customer orders.

2. The average order value is ₹1053.97.

3. Total revenue generated is ₹12,64,761.96.

4. Printer is the most ordered product, while Phone is the least ordered.

5. Product demand varies across categories, with Printer and Tablet showing the highest popularity.

6. Delivered orders represent successfully completed transactions.

7. SQL queries such as COUNT, AVG, SUM, GROUP BY, ORDER BY, and WHERE helped uncover key business insights from the dataset.